mimic/model/rackspace_image_store.py

```diff
@@ -0,0 +1,66 @@
+from six import iteritems
+from mimic.model.rackspace_images import (RackspaceWindowsImage, RackspaceArchImage,
+                                          RackspaceCentOSPVImage, RackspaceCentOSPVHMImage,
+                                          RackspaceCoreOSImage, RackspaceDebianImage,
+                                          RackspaceFedoraImage, RackspaceFreeBSDImage,
+                                          RackspaceGentooImage, RackspaceOpenSUSEImage,
+                                          RackspaceRedHatPVImage, RackspaceRedHatPVHMImage,
+                                          RackspaceUbuntuPVImage, RackspaceUbuntuPVHMImage,
+                                          RackspaceVyattaImage, RackspaceScientificImage,
+                                          RackspaceOnMetalCentOSImage, RackspaceOnMetalCoreOSImage,
+                                          RackspaceOnMetalDebianImage, RackspaceOnMetalFedoraImage,
+                                          RackspaceOnMetalUbuntuImage)
+
+
+@attributes([Attribute("image_list", default_factory=list)])
+class RackspaceImageStore(object):
+    """
+    A store for images to share between nova_api and glance_api
+    :var image_list: list of Rackspace images
+    """
+    def create_image_store(self, tenant_id):
+        """
+        Generates the data for each image in each image class
+        """
+        image_classes = [RackspaceWindowsImage, RackspaceArchImage, RackspaceCentOSPVImage,
+                         RackspaceCentOSPVHMImage, RackspaceCoreOSImage, RackspaceDebianImage,
+                         RackspaceFedoraImage, RackspaceFreeBSDImage, RackspaceGentooImage,
+                         RackspaceOpenSUSEImage, RackspaceRedHatPVImage, RackspaceRedHatPVHMImage,
+                         RackspaceUbuntuPVImage, RackspaceUbuntuPVHMImage, RackspaceVyattaImage,
+                         RackspaceScientificImage, RackspaceOnMetalCentOSImage,
+                         RackspaceOnMetalCoreOSImage, RackspaceOnMetalDebianImage,
+                         RackspaceOnMetalFedoraImage, RackspaceOnMetalUbuntuImage]
+        if len(self.image_list) < 1:
+            for image_class in image_classes:
+                for image, image_spec in iteritems(image_class.images):
+                    image_name = image
+                    image_id = image_spec['id']
+                    minRam = image_spec['minRam']
+                    minDisk = image_spec['minDisk']
+                    image_size = image_spec['OS-EXT-IMG-SIZE:size']
+                    image = image_class(image_id=image_id, tenant_id=tenant_id,
+                                        image_size=image_size, name=image_name, minRam=minRam,
+                                        minDisk=minDisk)
+                    if 'com.rackspace__1__ui_default_show' in image_spec:
+                        image.set_is_default()
+                    self.image_list.append(image)
+        return self.image_list
```

Comment:
This method has a side effect _and_ it has a return value; the return value is not based upon what the side effect was, but just whatever the current state happens to be after the side effect has taken place.  This is somewhat counter to the normal design of APIs within python; for example, consider `list.sort` - it doesn't return the list itself, so that you don't develop the mistaken impression that you can do `list.sort().reverse()` and not affect the original `list`; instead, there's a separate `sorted` function that does not modify the list in place but instead creates a copy.